### ISSUE TITLE
Incorrect style for printing of strings in Lists

### ISSUE THREAD

**Issue by poorna2152** [ballerina-platform/ballerina-lang#35883]
The spec states that a string can have either the expression style or the direct style.
https://ballerina.io/spec/lang/master/#ToString
So for the string: `"x\"y\\z"`,
- expression style: `"x\"y\\z"`
- direct style: `x"y\z`
```
import ballerina/io;

public function main() {
    any[] x = ["x\"y\\z"];
    io:println(x); // @output ["x"y\z"]
}
```

For the following program output by the compiler is `"x"y\z"` which is neither the expression style or the direct style. (surrounded by quotation marks).
I assume by the spec if a string is surrounded by quotation marks it should be in the expression style.

**Comment by BuddhiWathsala** [comment]
@poorna2152, here, you are not printing a string. Instead, you print an array of anydata. When we modified the code as below to print a string value, it printed the expected direct style output.

```ballerina
import ballerina/io;

public function main() {
    any[] x = ["x\"y\\z"];
    io:println(x); // @output ["x"y\z"]
    io:println(x[0]); // @output x"y\z
}
```

Actually, from the I/O printing perspective, we just get the string value of a given printable and print it to the standard out. I think @lasinicl can give some input about the string value extraction from the runtime perspective.

**Comment by poorna2152** [comment]
https://ballerina.io/spec/lang/master/#ToString
The table in the above link says that, when printing an array 
[X,Y]
X,Y are printed either in there expression style or the informal style. 
Since strings does not have a informal style I thought it should be the expression style that needs to be printed in this case. (which is `["x\"y\\z"]`)

**Comment by lasinicl** [comment]
I agree with @BuddhiWathsala, you are not printing a string, you are printing an array of anydata.
What you get when using I/O printing is the `direct style` mentioned in https://ballerina.io/spec/lang/master/#ToString.
It's the same as applying the `value:toString` on a value.

So if we go through the spec step by step,

x is an array. So you need to follow,
![arr](https://user-images.githubusercontent.com/19838787/164624568-d2c5e4f8-12bd-471c-b8b4-637d31141141.jpg)


Since it's the `direct` style that we need, the string representation of the array should be in the form `[X,Y]` where the array members (X, Y) are in their `informal` style. 
Your array has a `string` member. Therefore we need to consider the `informal` style of a `string` value. The spec has given the `informal` style for a `string` value as follows.

![str (1)](https://user-images.githubusercontent.com/19838787/164625706-8cd81ad9-c5b1-41ed-935e-c34852ac2a27.jpg)

That's why it prints the value as `["x"y\z"]`.
Hope this clarifies your question. Please let us know if you have any concerns.

**Comment by jclark** [comment]
Informal style for a string is the same as expression style, so the characters inside the string need escaping as in a string literal. It should be `["x\"y\\z"]`.

**Comment by jclark** [comment]
@MaryamZi There ought to be a test for this in the conformance test suite.
